src/map/packets/linkshell_concierge.cpp

```diff
@@ -0,0 +1,227 @@
+	case 2:
+	case 3:
+	case 4:
+
+        /*
+            Example SQL Query to use
+		    SELECT * FROM (
+            SELECT t1.listingid, t1.npcid, t1.linkshellid, t1.lslanguage, t1.lspearlcount, t1.lsactivedays, t1.lstimezone, t1.lstimeofday, t2.color, t2.name
+            FROM dspdb.linkshell_concierge t1 
+            INNER JOIN dspdb.linkshells t2 ON t1.linkshellid = t2.linkshellid ORDER BY t1.listingid DESC LIMIT 16
+            ) sub ORDER BY listingid ASC LIMIT 8,4 ;
+		*/
+        /*
+            Note: If you want to use a more visual way of listing this for like a webpage, or in your database viewer, you can use the following;
+                  This will however not list anything for non-existing Linkshells or Characters (the concierge itself still will, allowing you create a new shell-less linkshell)
+            SELECT t2.name, t3.charname, t1.*, t2.color, t3.accid FROM linkshell_concierge t1
+            INNER JOIN linkshells t2 ON t1.linkshellid = t2.linkshellid
+            INNER JOIN chars t3 ON t1.madebyplayerid = t3.charid ;
+        */
+
+		// Fill in as No results first
+		for (int8 i = 0; i < 4; i++)
+		{
+			uint8 slotOffset = ((option - 1) * 4) + i;
+			ref<uint8>(0x04 + i) = 0xFF; // List offset
+			ref<uint16>(0x08 + (i * 2)) = 0x0000; // Linkpearl icon
+			ref<uint16>(0x10 + (i * 0x18)) = 0x0000; // LinkShellID
+			ref<uint32>(0x70 + (i * 4)) = 0x00000000;
+		}
+
+        qStr = ("SELECT * FROM (");
+        // sql fields      0             1         2               3              4                5                6              7               8         9
+		qStr += "SELECT t1.listingid, t1.npcid, t1.linkshellid, t1.lslanguage, t1.lspearlcount, t1.lsactivedays, t1.lstimezone, t1.lstimeofday, t2.color, t2.name " ;
+		qStr += "FROM linkshell_concierge t1 INNER JOIN linkshells t2 ON t1.linkshellid = t2.linkshellid ";
+        qStr += " WHERE t1.lspearlcount > 0 ";
+
+		if (conciergeShareSettings == 0)
+		{
+            // Defaul setting, each NPC his own list
+			qStr += " AND t1.npcid = ";
+			qStr += std::to_string(conciergeID);
+		}
+        /*
+           Unused settings for conciergeShareSettings
+           Don't need to add a WHERE case for server-wide sharing setting
+		   if (conciergeShareSettings == 1)
+		   {
+		     qStr += " WHERE t1.npcid <> 0 ";
+		   }
+        */
+
+
+		qStr += " ORDER BY t1.listingid DESC LIMIT 16 "; // Only list 16 newest entries
+        qStr += ") sub ORDER BY listingid ASC "; // But showing them in the on the original order required the sub query
+        qStr += " LIMIT "; // limiting them
+		qStr += std::to_string((option -1) * 4); // starting point
+		qStr += ",4";
+		ret = Sql_Query(SqlHandle, qStr.c_str());
+
+		if (ret != SQL_ERROR && Sql_NumRows(SqlHandle) != 0 && Sql_NextRow(SqlHandle) == SQL_SUCCESS)
+        {
+			// Valid results
+			int i = 0;
+			while ((ret == SQL_SUCCESS) && (i < 4))
+			{
+                
+				uint16 lsID = Sql_GetUIntData(SqlHandle, 2);
```

Comment:
tabs to spaces please. 4 spaces per indent level.

Reply:
vs2017 has editorconfig support so you can automate it without fiddling with the vs settings https://visualstudiogallery.msdn.microsoft.com/c8bccfe2-650c-4b42-bc5c-845e21f96328